server/application/application.go

```diff
@@ -309,15 +300,69 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*v1
 		return newItems[i].Name < newItems[j].Name
 	})
 
+	if q.Offset != nil && q.Limit != nil {
+		if *q.Offset < 0 || *q.Limit < 0 {
+			return nil, fmt.Errorf("offset %d and limit %d must be a non-negative integer", *q.Offset, *q.Limit)
+		}
+		if *q.Offset >= int64(len(newItems)) || *q.Limit == 0 {
+			newItems = make([]v1alpha1.Application, 0)
+		} else {
+			if *q.Offset+*q.Limit >= int64(len(newItems)) {
+				newItems = newItems[*q.Offset:]
+			} else {
+				newItems = newItems[*q.Offset : *q.Offset+*q.Limit]
+			}
+		}
+	}
+
 	appList := v1alpha1.ApplicationList{
 		ListMeta: metav1.ListMeta{
 			ResourceVersion: s.appInformer.LastSyncResourceVersion(),
 		},
 		Items: newItems,
+		Stats: s.getAppsStats(apps),
```

Comment:
In the UI, we display a summary of the filtered data. So, should we keep it as usual?

https://github.com/user-attachments/assets/0c9836d8-4859-4a1e-90ca-8d126171fdd9

Reply:
I'm not quite sure. I'm also struggling with whether the statistics should be based on all the data or only the filtered data. It looks like the frontend can still perform statistics based on the filtered data, but what’s needed should be statistics based on all the data.

Reply:
From a frontend perspective, after adding pagination, we can no longer retrieve all app details, making it impossible to generate the correct summary from filtered data. Therefore, if filtering happens on the backend, the summary should also be calculated there.

Reply:
OK.... so, because of the pagination, I should return the "counters" based on the filtered data meanwhile the "items" of labels/namespaces/destinations are to be based on all data?